controllers/secret.go

```diff
@@ -45,6 +45,7 @@ func (r *IpfsReconciler) secretConfig(
 	if err != nil && errors.IsNotFound(err) {
 		// secret doesn't exist
 		generateNewIdentities(expectedSecret, 0, m.Spec.Replicas)
+		expectedSecret.Data = make(map[string][]byte)
```

Comment:
Doesn't this overwrite the map which we instantiated in `generateNewIdentities`? We should move this one line up